src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -2220,10 +2231,15 @@ test_cursor_implicit_session (void *ctx)
    ASSERT_POOL_SIZE (topology, 0);
    ASSERT_SESSIONS_MATCH (&test->sent_lsid, &find_lsid);
 
-   /* push a new server session into the pool */
+   /* push a new server session into the pool.  server session is only pushed
+    * if it is used.  therefore mark session as used prior to
+    * destroying session.  it isn't possible to use the server session
+    * to send a command because all messages must use the same lsid (for this
+    * test). */
```

Comment:
Looking into this test a bit more, notice how line 2245 resets the `lsid` that is captured in the APM events:
```
bson_reinit (&test->sent_lsid);
```
I believe the intention there is to bypass the assertion in the `started` callback, so the test can make assertions. This test is checking that a cursor ties to one session for all of the commands it sends. It checks that the same `lsid` is sent for the initial `find` command and subsequent `getMore` commands. It pushes a session into the pool in between the commands to ensure that the test doesn't pass just because the `getMore` obtains the same session from the pool.

I'd like to try once more for sending `ping` again to mark the session as used instead of overriding `last_used_usec`. I believe calling `bson_reinit (&test->sent_lsid);` before sending the `ping` should bypass the assertion.

If that doesn't work out, I'm ok with overriding `last_used_usec`. It is just a little less preferable to override since it is modifying a private field.